src/firehose.js

```diff
@@ -73,7 +74,14 @@ const FirehoseLogger = class FirehoseLogger extends Transport {
       message = Object.assign({ timestamp: (new Date()).toISOString() }, info);
       message = this.formatter(message);
     }
-    return this.firehoser.send(message);
+    this.firehoser
+      .send(message)
+      .then(() => {
+        this.emit('logged', message);
+      })
+      .catch((err) => {
+        this.emit('error', err);
+      });
```

Comment:
> One small hazard with this change is that if someone passed in their own Firehoser via the firehose option and its send() function didn't return a promise, this update would now cause an exception. So this update is technically a breaking API change. However, it looks like it was always expected that the Firehoser's send() call returns a promise from the base class documentation here. The code prior to the Winston 3 upgrade also expected a promise here.

My only comment is w.r.t. this statement, maybe there are two options here:
1.  In your view would it make sense to just return this `Promise` in order to avoid a potentially breaking change? I think you're right that practically nobody is expecting to use it this way
2. If not, would you mind just adjusting the commented annotation in `IFirehoser` [here](https://github.com/pkallos/winston-firehose/blob/master/src/firehose.js#L33) to indicate this no longer returns a promise?

With option 1, I will deploy as a minor-level version change,
With option 2, I will bump the major version.

Reply:
I think we should go with option #2: to update the API and release a breaking change. If we went with option #1 and returned the promise from `log()`, that promise would no longer throw an error (because it'd crash the process). So that option would technically be a breaking change also. We might as well go with #2.

Regarding updating the comment [here](https://github.com/pkallos/winston-firehose/blob/master/src/firehose.js#L33), that function still needs to return a promise (that's caught [here](https://github.com/mixmaxhq/winston-firehose/blob/handle-errors/src/firehose.js#L79)).

Reply:
Alright, sounds good then I agree.

>Regarding updating the comment here, that function still needs to return a promise (that's caught here).

You're right, my bad.

Will merge and publish v3.0.0.